lib/cocoapods-core/specification/root_attribute_accessors.rb

```diff
@@ -47,7 +47,7 @@ def swift_version
         #
         def swift_versions
           @swift_versions ||= begin
-            swift_versions = Array(attributes_hash['swift_versions'])
+            swift_versions = Array(attributes_hash['swift_versions']).dup
```

Comment:
further below we do:
```ruby
swift_versions << attributes_hash['swift_version'] unless attributes_hash['swift_version'].nil?
```

At this point `swift_versions` points to the original array and it actually alters the internal structure of the specification.

Reply:
tbh I didn't realize this would ever alter the original structure. TIL.

Reply:
Yeap `Array()` doesn't make a new one if what we are parsing is already an array....🤦‍♂